lib/entities/resource-provider.ts

```diff
@@ -92,6 +105,44 @@ function createResourceProviderApi(makeRequest: MakeRequest) {
         params: getParams(data),
       })
     },
+
+    getResourceType: function getResourceType(id: string) {
+      return makeRequest({
+        entityType: 'ResourceType',
+        action: 'get',
+        params: {
+          organizationId: this.sys.organization.sys.id,
+          appDefinitionId: this.sys.appDefinition.sys.id,
+          resourceTypeId: id,
+        },
+      }).then((data) => wrapResourceType(makeRequest, data))
+    },
+    upsertResourceType: function upsertResourceType(id: string, data: UpsertResourceTypeProps) {
+      return makeRequest({
+        entityType: 'ResourceType',
+        action: 'upsert',
+        params: {
+          organizationId: this.sys.organization.sys.id,
+          appDefinitionId: this.sys.appDefinition.sys.id,
+          resourceTypeId: id,
+        },
+        headers: {},
```

Comment:
Was surprised to see headers as empty object here, since headers is optional. Just entirely out of curiosity, do you know if this was required for some reason?

Reply:
you are right, yeah, we can remove it 👍